src/python/py_lora_adapter.cpp

```diff
@@ -0,0 +1,91 @@
+// Copyright (C) 2023-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include <filesystem>
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+#include <pybind11/stl_bind.h>
+#include <pybind11/functional.h>
+
+#include "./utils.hpp"
+
+namespace py = pybind11;
+namespace utils = ov::genai::pybind::utils;
+
+
+void init_lora_adapter(py::module_& m) {
+    py::class_<ov::genai::Adapter>(m, "Adapter", "Inmutable LoRA Adapter that carries the adaptation matrices and serves as unique adapter identifier.")
+        .def(py::init([](
+            const std::string& path
+        ) {
+            return std::make_unique<ov::genai::Adapter>(path);
```

Comment:
The `Adapter` class already behaves as a smart pointer like `std::shared_ptr`. Do we need an extra level of indirection with `unique_ptr`?

Reply:
Agree, not needed. Removed `unique_ptr`.